spec/webauthn/authenticator_attestation_response_spec.rb

```diff
@@ -54,48 +127,70 @@
+      let(:origin) { "http://localhost:3000" }
 
-      WebAuthn::AuthenticatorAttestationResponse.new(
-        attestation_object: Base64.strict_decode64(response[:attestation_object]),
-        client_data_json: Base64.strict_decode64(response[:client_data_json])
-      )
-    end
+      let(:attestation_response) do
+        response = seeds[:security_key_direct][:authenticator_attestation_response]
 
-    before do
-      WebAuthn.configuration.attestation_root_certificates_finders = finder_for('feitian_ft_fido_0200.pem')
-    end
+        WebAuthn::AuthenticatorAttestationResponse.new(
+          attestation_object: Base64.strict_decode64(response[:attestation_object]),
+          client_data_json: Base64.strict_decode64(response[:client_data_json])
+        )
+      end
 
-    it "verifies" do
-      expect(attestation_response.verify(original_challenge)).to be_truthy
-    end
+      before do
+        WebAuthn.configuration.attestation_root_certificates_finders = finder_for('feitian_ft_fido_0200.pem')
+      end
 
-    it "is valid" do
-      expect(attestation_response.valid?(original_challenge)).to eq(true)
-    end
+      it_behaves_like "is valid and verifies"
 
-    it "returns attestation info" do
-      attestation_response.valid?(original_challenge)
+      it "returns attestation info" do
+        attestation_response.valid?(original_challenge)
 
-      expect(attestation_response.attestation_type).to eq("Basic_or_AttCA")
-      expect(attestation_response.attestation_trust_path).to all(be_kind_of(OpenSSL::X509::Certificate))
-    end
+        expect(attestation_response.attestation_type).to eq("Basic_or_AttCA")
+        expect(attestation_response.attestation_trust_path).to all(be_kind_of(OpenSSL::X509::Certificate))
+      end
 
-    it "returns the credential" do
-      expect(attestation_response.credential.id.length).to be >= 16
+      it "returns the credential" do
+        expect(attestation_response.credential.id.length).to be >= 16
+      end
+
+      it "returns the attestation certificate key" do
+        expect(attestation_response.attestation_certificate_key_id).to(
+          eq("f4b64a68c334e901b8e23c6e66e6866c31931f5d")
+        )
+      end
     end
 
-    it "returns the attestation certificate key" do
-      expect(attestation_response.attestation_certificate_key_id).to(
-        eq("f4b64a68c334e901b8e23c6e66e6866c31931f5d")
-      )
+    context "when there are multiple allowed origins" do
+      include_context "with multiple origins"
+
+      context "when rp_id is set explicitly" do
+        include_context "with rp_id set to", "localhost"
+
+        it_behaves_like "is valid and verifies"
+      end
+
+      context "when rp_id is not set explicitly" do
+        include_context "with rp_id set to", nil
+
+        it "verifies" do
+          expect { attestation_response.verify(original_challenge) }.to raise_error(WebAuthn::RpIdVerificationError)
+        end
```

Comment:
We should change the name of the spec in the cases where it raises an exception, right?